core/base/array.cpp

```diff
@@ -92,9 +94,42 @@ void Array<ValueType>::fill(const ValueType value)
 }
 
 
+template <typename ValueType>
+void reduce(const Array<ValueType>& input_arr, ValueType* value)
+{
+    auto exec = input_arr.get_executor();
+    exec->run(array::make_reduce_array(input_arr.get_const_data(),
+                                       input_arr.get_num_elems(), value));
+}
+
+
+template <typename ValueType>
+ValueType reduce(const Array<ValueType>& input_arr, const ValueType init_value)
+{
+    auto exec = input_arr.get_executor();
+    auto value = Array<ValueType>(exec, {init_value});
+    exec->run(array::make_reduce_array(input_arr.get_const_data(),
+                                       input_arr.get_num_elems(),
+                                       value.get_data()));
+    return exec->copy_val_to_host(value.get_data());
```

Comment:
It can be done with normal reduction and report init_value + result?
the normal reduction do not have additional init in the beginning

Reply:
Yes, the idea of having a host return was why this function was added.